lib/rules/base.js

```diff
@@ -524,12 +515,25 @@ module.exports = class Base {
       rule: this.ruleName,
       severity: this.severity,
     };
+
     if (this._filePath) {
       defaults.filePath = this._filePath;
     }
+
     if (this[MODULE_NAME]) {
       defaults.moduleId = this[MODULE_NAME];
     }
+
+    if (result.node) {
+      let node = result.node;
+
+      result.line = node.loc && node.loc.start.line;
+      result.column = node.loc && node.loc.start.column;
+      result.source = this.sourceForNode(node);
+
+      delete result.node;
```

Comment:
Realized reading #1771 that we should be deleting this so the result is stable (thanks @josephdsumner!).